python/dgl/graphbolt/itemset.py

```diff
@@ -151,11 +151,10 @@ def __getitem__(self, idx: Union[int, slice, Iterable]) -> Tuple:
                 f"{type(self).__name__} instance doesn't support indexing."
             )
         if isinstance(self._items, int):
-            assert isinstance(idx, (int, torch.Tensor)), (
-                f"Indexing of integer-initialized {type(self).__name__} "
-                f"instance must be int or torch.Tensor."
-            )
             # [Warning] Index range is not checked.
```

Comment:
Does this warning still apply?

Reply:
yes, it's applied to `return idx`

Reply:
How hard to apply an filter?

Reply:
added for `int` and remove support for `Tensor` as `slice` and `int` are the ways we index in `ItemSampler` for now. If we need more indexing ways, we enhance later. Adding support `Tensor` requires more checks such as range, dtype.